core/state/snapshot/snapshot.go

```diff
@@ -379,6 +388,20 @@ func (t *Tree) Cap(root common.Hash, layers int) error {
 	if !ok {
 		return fmt.Errorf("snapshot [%#x] is disk layer", root)
 	}
+
+	// Update the cache stats
+	if diff.origin != nil && diff.origin.cache != nil {
```

Comment:
It's very weird to upload the cache stats here.

The associated disk layer will only be modified if the size bottom-most diff layer reaches the threshold, it's not always be updated...